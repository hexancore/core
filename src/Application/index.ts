export * from './HcApplicationModule';
export * from './GeneralBus';
export * from './DefaultGeneralBus';
