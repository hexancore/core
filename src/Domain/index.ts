export * from './Entity/index';
export * from './Repository/index';
