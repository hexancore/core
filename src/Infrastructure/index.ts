export * from './Persistence/Domain';
export * from './Config';
export * from './Account';
