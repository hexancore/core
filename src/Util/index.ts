export * from './Log';
export * from './Filesystem/FileTypeReadable';

