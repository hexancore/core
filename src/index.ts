export * from './Application/index';
export * from './Domain/index';
export * from './Infrastructure/index';
export * from './Util/index';

