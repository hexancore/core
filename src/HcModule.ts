import { ConfigurableModuleBuilder, Global, Module } from '@nestjs/common';
import { CurrentTime } from '@hexancore/common';
import { HcAppConfigModule } from './Infrastructure/Config/HcAppConfigModule';
import { HcApplicationModule } from './Application';
import { EntityPersisterFactoryManager, HcAppRedisModule } from './Infrastructure';
import { ClsModule } from 'nestjs-cls';
import { nanoid } from 'nanoid';
import { AccountContext } from './Infrastructure/Account/AccountContext';

export interface HcModuleOptions {}

export interface HcModuleExtras {
  cls?: boolean;
  redis?: boolean;
}

export const { ConfigurableModuleClass, MODULE_OPTIONS_TOKEN } = new ConfigurableModuleBuilder<HcModuleOptions>()
  .setClassMethodName('forRoot')
  .setExtras<HcModuleExtras>({ cls: false, redis: false }, (def, extras) => {
    def.imports = def.imports ? def.imports : [];
    def.providers = def.providers ? def.providers : [];
    def.exports = def.exports ? def.exports : [];

    if (extras.cls) {
      def.imports.push(
        ClsModule.forRoot({
          global: true,
          middleware: { mount: false, generateId: true, idGenerator: (req) => req.headers['X-Request-Id'] ?? nanoid(21) },
        }),
      );
      def.providers.push(AccountContext);
      def.exports.push(AccountContext);
    }

    if (extras.redis) {
      def.imports.push(HcAppRedisModule);
    }
    return def;
  })
  .build();

@Global()
@Module({
  imports: [HcAppConfigModule, HcApplicationModule],
  providers: [
    {
      provide: CurrentTime,
      useFactory: () => new CurrentTime(),
    },
    EntityPersisterFactoryManager,
  ],
  exports: [CurrentTime, EntityPersisterFactoryManager],
})
export class HcModule extends ConfigurableModuleClass {}
