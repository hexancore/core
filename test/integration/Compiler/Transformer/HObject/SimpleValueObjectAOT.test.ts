/**
 * @group integration
 */

import { FeatureTsTransformerTestHelper } from "@test/FeatureTsTransformerTestHelper";
import { BookId } from "@test/libs/test-lib/src";

describe("HObject.SimpleValueObject", () => {
  let helper: FeatureTsTransformerTestHelper;

  beforeAll(() => {
    helper = FeatureTsTransformerTestHelper.create();
  });

  test("transform", () => {
    const sourceFilePath = `/Book/Domain/Book/Shared/ValueObject/BookId.ts`;
    const out = helper.transform(sourceFilePath);

    expect(out).toMatchSnapshot();
  });

  describe("parse", () => {
    test("when valid", () => {
      const current = BookId.parse(10);

      expect(current).toMatchSnapshot();
    });
    test("when invalid", () => {
      const current = BookId.parse("invalid");

      expect(current).toMatchSnapshot();
    });
  });

  test("toJSON", () => {
    const current = BookId.cs(10);

    expect(JSON.stringify(current, undefined, 2)).toMatchSnapshot();
  });
});


