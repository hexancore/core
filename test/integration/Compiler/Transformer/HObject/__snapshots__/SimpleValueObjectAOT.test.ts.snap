// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HObject.SimpleValueObject parse when invalid 1`] = `
Result {
  "value": {
    "cause": null,
    "code": 400,
    "data": {
      "code": "invalid_hobject",
      "i18n": "core.plain.parse_issue.invalid_hobject",
      "issues": [
        {
          "code": "invalid_type",
          "expected": "uint",
          "i18n": "core.plain.parse_issue.invalid_type",
          "message": "Expected value type: 'uint', received: 'string'",
          "path": undefined,
          "received": "string",
        },
      ],
      "message": "Invalid plain object to parse to HObject: book.domain.book.value_object.book_id",
      "path": undefined,
      "typeId": "book.domain.book.value_object.book_id",
    },
    "error": null,
    "i18n": "",
    "message": "",
    "type": "core.plain.parse",
  },
}
`;

exports[`HObject.SimpleValueObject parse when valid 1`] = `
Result {
  "value": 10,
}
`;

exports[`HObject.SimpleValueObject toJSON 1`] = `"10"`;

exports[`HObject.SimpleValueObject transform 1`] = `
"import { UInt, R, OK, PlainParseHelper, IntegerPlainParseHelper, ArrayPlainParseHelper, StringPlainParseHelper, NumberPlainParseHelper, HObjectTypeMeta, type JsonObjectType, type PlainParseError, type PlainParseIssue, type AnyValueObject, type ValueObjectType } from "@hexancore/common";
export class BookId extends UInt<BookId> {
    public static HOBJ_META = HObjectTypeMeta.domain("Book", "Book", "ValueObject", "BookId", BookId);
}
"
`;
